app_rules/generate_app_rules.py

```diff
@@ -95,63 +96,75 @@ def __init__(self, app_rules, app_name):
 		self.app_rules = app_rules
 
 	def generate_rules(self):
-		with open(OUTFILE, 'a') as o:
+		with open(OUTFILE, "a") as o:
 			o.write("".join(["\n", TAB, COMMENT, self.app_name, "\n"]))
 		for r in self.app_rules:
 			IgnoreRule(r).add_rule()
 
 
 class IgnoreRule:
+	_implemented = ("Equals", "StartsWith", "EndsWith", "Contains")
+	_processed = []
+
 	def __init__(self, rule):
 		self.kind = rule["kind"]
 		self.id = rule["id"]
-		self.matching_strategy = rule[_] if (_ := "matching_strategy") in rule else None
+		self.matching_strategy = rule[_] if (_ := "matching_strategy") in rule else "Legacy"
 		self.comment = rule[_] if (_ := "comment") in rule else None
 
-		# Raise error for unsupported matching strategies
-		# If future rules use regex, we can implement them via "AddTitleMatchFilter"
-		if self.matching_strategy and self.matching_strategy != "Equals":
-			raise RuntimeError('Matching strategy "{_}" unsupported')
+		if self.kind not in ("Class", "Exe", "Title"):
+			raise RuntimeError("Undefined kind: " + self.kind)
+
+		# convert legacy to new matching strategy
+		if self.matching_strategy == "Legacy":
+			self.matching_strategy = "Equals" if self.kind == "Exe" else "Contains"
```

Comment:
From what I can tell in komorebi, the default matching strategy is `Equals` not `Contains`:

https://github.com/LGUG2Z/komorebi/blob/ddfcf8b76ffb0fdc7888eaba9c6dcf22ca5603c0/komorebi-core/src/config_generation.rs#L102-L111

If I'm wrong, then having then the following would mean if you opened say, a Word documented with the text `Calculator`, the window would be ignored:

https://github.com/urob/Whim/blob/a54e4284b88bdbab1299746be85859e46677040e/src/Whim/Filter/DefaultFilteredWindowsKomorebi.g.cs#L62-L63

Reply:
> From what I can tell in komorebi, the default matching strategy is `Equals` not `Contains`:
> 
> 
> 
> https://github.com/LGUG2Z/komorebi/blob/ddfcf8b76ffb0fdc7888eaba9c6dcf22ca5603c0/komorebi-core/src/config_generation.rs#L102-L111
> 

I thought the replacement happens only for `Exe` here. For `Class` and `Title`, there is still a `Legacy` property which later expands to `OpensWith || EndsWith` if I read this correctly:

https://github.com/LGUG2Z/komorebi/blob/ddfcf8b76ffb0fdc7888eaba9c6dcf22ca5603c0/komorebi/src/window.rs#L561:L577

I initially thought that `Contains` is close enough, but that might have been too eager. 

From looking at the existing legacy rules, my impression is that almost all -- if not all of them -- are actually matching on `Equals`. If we want to merge something quickly we could set Legacy to Equals for now and keep an eye out if there is any cases where this is too strict. 

I'll also try to reach out on the komorebi discord and ask for clarification.

Reply:
I think you're right and the behavior matches. However, I'm a bit wary of being so lax with `Title`, as it could hit other cases like a browser window searching for a program. `Class` I have less of an issue with - I think unintentional behavior is less likely, but even so, it's a bit unwise in my opinion.

> I'll also try to reach out on the komorebi discord and ask for clarification.

I'd appreciate this - it'd be interesting to see what they think about this.

Reply:
> On the other hand, komorebi is widely used. If there were too many problems, it probably would have already come to surface.

I think there's validity to this. It also makes me think of how users could turn off specific rules, which isn't possible right now. I added #702 to accommodate this in the future, which would make it possible to add to your config the ability to ignore overzealous rules.

Reply:
> Let me know what you think. If you agree with `StartsWith || EndsWith` I would set it up as follows

Let's go with that and see what happens - _should_ be fine 🤞

Reply:
> > On the other hand, komorebi is widely used. If there were too many problems, it probably would have already come to surface.
> 
> 
> 
> I think there's validity to this. It also makes me think of how users could turn off specific rules, which isn't possible right now. I added #702 to accommodate this in the future, which would make it possible to add to your config the ability to ignore overzealous rules.

Yea, that would be nice.

Reply:
> > Let me know what you think. If you agree with `StartsWith || EndsWith` I would set it up as follows
> 
> Let's go with that and see what happens - _should_ be fine 🤞

Done. New test run is here: https://github.com/urob/Whim/pull/3/files